vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/Grid.java

```diff
@@ -527,6 +527,50 @@ public Renderer<T> getRenderer() {
             return renderer;
         }
 
+        /**
+         * Switch the renderer used for this column.
+         *
+         * @param renderer
+         *            the new renderer to be used for this column, should never
+         *            be {@code null}
+         *
+         * @since 24.1
+         */
+        public void setRenderer(Renderer<T> renderer) {
```

Comment:
If the renderer is set _after_ calling `column.setEditorComponent(field)`, the editor doesn't render:

Example (used `GridViewEditorPage.java`):
```java
nameColumn.setEditorComponent(field);

// If I change the renderer _after_ setting the editor component, the
// editor for the column is rendered as "null"
nameColumn.setRenderer(LitRenderer.<Person>of("<b>${item.firstName}</b>")
                .withProperty("firstName", item -> item.getFirstName()));
```

https://user-images.githubusercontent.com/1222264/233973974-2a7c7d5a-19a2-49ed-a40b-c155a3e3a84b.mp4

Reply:
Seems related to calling `destroyDataGenerators` when setting the renderer, which also removes the editor data generator.